net/packet/af_packet.c

```diff
@@ -1713,19 +1713,17 @@ static int fanout_add(struct sock *sk, u16 id, u16 type_flags)
 	    match->prot_hook.type == po->prot_hook.type &&
 	    match->prot_hook.dev == po->prot_hook.dev) {
 		err = -ENOSPC;
-		if (atomic_read(&match->sk_ref) < PACKET_FANOUT_MAX) {
+		if (refcount_read(&match->sk_ref) < PACKET_FANOUT_MAX) {
 			__dev_remove_pack(&po->prot_hook);
 			po->fanout = match;
-			po->rollover = rollover;
-			rollover = NULL;
-			atomic_inc(&match->sk_ref);
```

Comment:
The two statements above this line should stay to match the diff of the patch: https://github.com/android-linux-stable/msm-4.4/commit/171588e9ad56705b9bd1b1af80c0467063937d0d

This also matched mainline: https://github.com/torvalds/linux/blob/master/net/packet/af_packet.c#L1752-L1760

Reply:
Fixed this one now